sql: don't escape names in crdb_internal rows

madelynnblue opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13854)
<!-- Reviewable:end -->

knz commented:
LGTM, however please add a little context/motivation in the commit message.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13854#-:-Ke5PyDpe0u2q3k4pyR2:bget8nv)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: don't escape names in crdb_internal rows

Previously the table names would be escaped which would permit them to
be usable in SQL strings if there were an external program crafting
these SQL strings manually. However these escaped names would be
incorrect if used in placeholders or in subqueries or joins comparing
table names.
